lib/timberline/worker.rb

```diff
@@ -48,6 +56,24 @@ def keep_watching?
       true
     end
 
+    # Whether the worker is currently executing a job.
+    #
+    # @return [boolean]
+    def executing_job?
+      @executing_job == true
+    end
+
+    # Called when process_item has resulted in an exception. Exceptions are
+    # handled by default by reraising. This method is available to allow
+    # subclasses to override or extend how exceptions are handled. A subclass
+    # may, for example, want to log errors, handle specific error types in
+    # specific ways, or call error_item by default on all errors.
+    #
+    # @raise By default, raises the error it is provided.
+    def handle_process_exception(exception, item)
```

Comment:
Unused method argument - `item`. If it's necessary, use `_` or `_item` as an argument name to indicate that it won't be used.